marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mastering/calculateMergingActivityLib.sjs

```diff
@@ -0,0 +1,54 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+const currentDatabaseName = xdmp.databaseName(xdmp.database());
+
+function calculateMergingActivity(step)
+{
+    let sourceDatabaseName = step.sourceDatabase || currentDatabaseName;
+    if (sourceDatabaseName !== currentDatabaseName) {
+        const HubUtils = require("/data-hub/5/impl/hub-utils.sjs");
+        return new HubUtils().queryLatest(function () {
+            return internalCalculateMergingActivity(step);
+        }, sourceDatabaseName);
+    } else {
+        return internalCalculateMergingActivity(step);
+    }
+}
+
+function internalCalculateMergingActivity(step)
+{
+    const targetEntityType = step.targetEntity || step.targetEntityType;
+    let query = null;
+    if (targetEntityType) {
+        let entityTypeTitle = targetEntityType;
+        if (targetEntityType.includes('/')) {
+            entityTypeTitle = require("/data-hub/5/impl/entity-lib.sjs").getEntityTypeIdParts(targetEntityType).entityTypeTitle;
+        }
+        query = cts.orQuery([
```

Comment:
Pinging @rahulvudutala  - this query is saying "I can find instances of an entity type by querying on 'title' or 'es:title'", while the Explore feature is saying "I can find instances of an entity type by querying on a collection with the same name as the entity type title". We should have one way of querying for entities of a particular type - what's the right answer?